### DISCUSSION TITLE
Why transaction does not rollback while inserting row with raw SQL?

### DISCUSSION THREAD

**Discussion by kom24al** [sequelize/sequelize#17237] [Help & Questions] [answered]
await db.sequelize.transaction(async (transaction) => {
    const [orderInfoNew] = await Promise.all([
	

	// Execute the INSERT query
	await db.sequelize.query(insertQuery, {
	  replacements: orderInfoData,
	  type: db.Sequelize.QueryTypes.INSERT
	},{ transaction: transaction })
])
})

**Comment by ephys** [comment]
Transactions roll back when an error is thrown in the transaction callback

**Reply by kom24al** [reply]
@ephys Can you please tell, what is wrong with the above code that when subsequent transaction fails the it is not getting rollbacked?
Transaction rollback is only working with sequelize.OrderInfo.create method not with RAW query. My database is in Oracle.

**Reply by ephys** [reply]
You're passing the transaction as a third parameter but this function only takes two parameters: the SQL and the options

**Reply by kom24al** [reply]
Is there any other way to manage transaction with sequelize.query?

**Reply by ephys** [reply, marked as answer]
Pass the transaction option in the second parameter:

```ts
await sequelize.query(insertQuery, {
  replacements: orderInfoData,
  type: QueryTypes.INSERT,
  transaction,
})
```
